examples/godot_tps_controller_port/src/player.rs

```diff
@@ -0,0 +1,814 @@
+            {
+                let shoot_origin = unsafe {
+                    self.shoot_from
+                        .unwrap()
+                        .assume_safe()
+                        .global_transform()
+                        .origin
+                };
+
+                let ch_pos = unsafe {
+                    self.crosshair.unwrap().assume_safe().get_rect().position
+                        + self.crosshair.unwrap().assume_safe().get_rect().size * 0.5
+                };
+                let ray_from = unsafe {
+                    self.camera_camera
+                        .unwrap()
+                        .assume_safe()
+                        .project_ray_origin(ch_pos)
+                };
+                let ray_dir = unsafe {
+                    self.camera_camera
+                        .unwrap()
+                        .assume_safe()
+                        .project_ray_normal(ch_pos)
+                };
+
+                let mut shoot_target = Vector3::default();
+
+                let raycast_exclusion_array = VariantArray::new();
+                raycast_exclusion_array.push(unsafe { owner.assume_shared() });
+
+                let col = unsafe {
+                    owner
+                        .get_world()
+                        .unwrap()
+                        .assume_safe()
+                        .direct_space_state()
+                        .unwrap()
+                        .assume_safe()
+                        .intersect_ray(
+                            ray_from,
+                            ray_from + ray_dir * 1000.0,
+                            raycast_exclusion_array.into_shared(),
+                            0b11,
+                            true,
+                            false,
+                        )
+                };
+
+                godot_print!("{:?}", col);
+
+                if col.is_empty() {
+                    shoot_target = ray_from + ray_dir * 1000.0;
+                } else {
+                    shoot_target = col.values().get(0).to().unwrap();
+                }
+                let shoot_dir = (shoot_target - shoot_origin).normalized();
+
+                let bullet = unsafe {
+                    load::<PackedScene>("res://player/bullet/bullet.tscn")
+                        .unwrap()
+                        .assume_safe()
+                        .instance(0)
+                        .unwrap()
+                        .assume_safe()
+                        .cast::<KinematicBody>()
+                        .unwrap()
+                };
+                owner.add_child(bullet, true);
+                bullet.global_transform().origin = shoot_origin;
+                // If we don't rotate the bullets there is no useful way to control the particles.
+                bullet.look_at(shoot_origin + shoot_dir, Vector3::UP);
+                bullet.add_collision_exception_with(bullet.get_parent().unwrap());
```

Comment:
Same beginner mistake as I explained on Discord. There are no implicit references in Rust. You're just modifying a value and throwing it away. I'm curious why Rust isn't emitting a warning for you here though. Are you sure you aren't accidentally filtering anything out in your editor? Or did some of our `#[allow]` attributes leak through to user code?

Reply:
It could be although I'm not completely sure. Also #975 changes a lot how the macros work under the hood, so it isn't worth much investigating on `master` right now.

Reply:
Wow. That's a find! My eyes have been glued to the compiler output and Clippy for long enough that I've just assumed I was alright regarding problems of this sort in the absence of complaints. Thanks for the pointer.